packages/subscription/src/stream.ts

```diff
@@ -114,6 +119,28 @@ export function createStream<T, TReturn = undefined>(callback: Callback<T, TRetu
       });
     },
 
+    stringBuffer(scope: Task): StringBufferStream<string, TReturn> {
+      let buffer = "";
+
+      scope.spawn(subscribable.forEach((m) => { buffer += `${m}` }));
+
+      let stream = createStream<string, TReturn>((publish) => function*() {
+        let internalBuffer = buffer;
+        publish(internalBuffer);
+        return yield subscribable.forEach((m: T) => {
+          internalBuffer += `${m}`;
+          publish(internalBuffer);
+        });
+      });
+
+      return {
+        ...stream,
+        get value(): string {
+          return buffer;
+        }
```

Comment:
How do we even use this from TypeScript?

Reply:
It uses an extended interface, see here: https://github.com/thefrontside/effection/pull/255/files#diff-0fd53fb3d32144b53a5c547ff0713daaee9dd8428e8502bfb9a8f0b720f7e573R23-R25

I have some thoughts about this though, I think there might be a more comprehensive/powerful solution available here. We could have something like a `ValueStream<T>` which is more general than a type specific for strings. Some methods on this, like `map` could then return a new value stream. We could also add a `unique` method on this type of stream. We would be able to implement this interface not just for buffered streams, but also for Atom for example.

Reply:
Interestingly this leads back to an idea I had a long, long time ago about client side frameworks. This concept is similar to Ember Octanes change tracking, and even before that came out I was tinkering with that. If you squint a little bit it's also similar to React's `useState` hook, except a lot less weird ;)

Reply:
Right, but the `Stream#stringBuffer` interface returns a `Stream`. The test itself never uses the `value` property, only the accumulated value on the stream iterator. It's not a biggie, but I don't see a way that we can actually get a reference to a `StringBufferStream` without down-casting.

Reply:
Oh duh! I totally missed that, and yes we should totally have a test for this too!